### DISCUSSION TITLE
Help message has extra backslashes

### DISCUSSION THREAD

**Discussion by seowalex** [fastapi/typer#1058] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer

app = typer.Typer(rich_markup_mode=None)

@app.command()
def main(name: str):
    typer.echo(f"Hello {name}")

if __name__ == "__main__":
    app()
```

### Description

When `rich_markup_mode` is set to `None`, the help message has extra backslashes:

```
Arguments:
  NAME  \[required]
```

This seems to have been caused by the escaping done in https://github.com/fastapi/typer/commit/5996fc18cb4cf29854b592061581f42957ce6257 when Rich is detected. However, in my case, even though I have Rich installed, I have explicitly set `rich_markup_mode` to `None`.

### Operating System

Linux

### Typer Version

0.13.1

### Python Version

3.12.7

**Comment by svlandeg** [comment]
Thanks for the report, we'll look into it! 🙏

**Reply by seowalex** [reply]
This seems to still be present in latest version.

**Reply by svlandeg** [reply]
That's most likely true, as we haven't had time to look into this one yet, but it's definitely high on my queue! Thanks for your patience 🙏

**Reply by seowalex** [reply]
No worries, just checking in! Typer is fantastic to use otherwise.

**Comment by svlandeg** [comment]
As an update: I've been able to reproduce this, track it back to the PR you mentioned, and created a test to capture the correct behaviour. Now looking into fixing this while keeping the intended functionality from #847 as well.

**Comment by svlandeg** [comment, marked as answer]
PR is up here: https://github.com/fastapi/typer/pull/1089
